packages/core/src/lib/handler/sms.handler.spec.ts

```diff
@@ -28,8 +29,36 @@ test('send sms should call the provider method correctly', async () => {
 
   expect(spy).toHaveBeenCalled();
   expect(spy).toHaveBeenCalledWith({
-    'content': 'Name: test name',
-    'to': '+1333322214'
+    content: 'Name: test name',
+    to: '+1333322214',
   });
   spy.mockRestore();
+
+  const spyTemplateFunction = jest
+    .fn()
+    .mockImplementation(() => Promise.resolve('test'));
+
+  emailHandler = new SmsHandler(
+    {
+      subject: 'test',
+      channel: ChannelTypeEnum.SMS,
+      template: spyTemplateFunction,
+    },
+    provider
+  );
+
+  await emailHandler.send({
+    $email: 'test@email.com',
+    $phone: '+1333322214',
+    $user_id: '1234',
+    firstName: 'test name',
+  });
+
+  expect(spyTemplateFunction).toHaveBeenCalled();
+  expect(spyTemplateFunction).toBeCalledWith({
+    $email: 'test@email.com',
+    $phone: '+1333322214',
+    $user_id: '1234',
+    firstName: 'test name',
+  });
```

Comment:
@davidsoderberg Maybe it's better to create a new `test` case to specifically test the template function?